source/server/active_tcp_listener.cc

```diff
@@ -214,8 +214,9 @@ void ActiveTcpSocket::newConnection() {
 
 void ActiveTcpListener::onAccept(Network::ConnectionSocketPtr&& socket) {
   if (listenerConnectionLimitReached()) {
-    ENVOY_LOG(trace, "closing connection: listener connection limit reached for {}",
-              config_->name());
+    ASSERT(socket->addressProvider().remoteAddress() != nullptr);
```

Comment:
I should have checked this earlier, but per STYLE.md this should be a RELEASE_ASSERT, I think. It explicitly calls out this case as somewhere we shouldn't just ASSERT.

Reply:
A drive by: As this case is going to crash in an obvious way on the next line I think in general we probably don't need any assert here, but if we want to document our assumption I think a normal ASSERT is fine here. cc @zuercher @rgs1. I might be misunderstanding the new style guide though.